lib/plugins/aws/package/compile/events/apiGateway/lib/resources.test.js

```diff
@@ -242,4 +250,194 @@ describe('#compileResources()', () => {
+        .Resources.ApiGatewayResourceFooBar.Properties.ParentId)
+        .to.equal('axcybf2i39');
+      expect(awsCompileApigEvents.serverless.service.provider.compiledCloudFormationTemplate
+        .Resources.ApiGatewayResourceBarIdVar.Properties.ParentId.Ref)
+        .to.equal('ApiGatewayResourceBar');
+      expect(awsCompileApigEvents.serverless.service.provider.compiledCloudFormationTemplate
+        .Resources.ApiGatewayResourceUsersMePosts).not.equal(undefined);
+      expect(awsCompileApigEvents.serverless.service.provider.compiledCloudFormationTemplate
+        .Resources.ApiGatewayResourceUsersFriendsComments.Properties.ParentId)
+        .to.equal('fcasdoojp1');
+    });
+  });
+
+  it('should not create any child resources if all resources exists', () => {
+    awsCompileApigEvents.serverless.service.provider.apiGateway = {
+      restApiId: '6fyzt1pfpk',
+      restApiRootResourceId: 'z5d4qh4oqi',
+      restApiResources: {
+        foo: 'axcybf2i39',
+        users: 'zxcvbnmasd',
+        'users/friends': 'fcasdoojp1',
+        'users/is/this/a/long/path': 'sadvgpoujk',
+      },
+    };
+
+    awsCompileApigEvents.validated.events = [
+      {
+        http: {
+          path: 'foo',
+          method: 'GET',
+        },
+      },
+      {
+        http: {
+          path: 'users',
+          method: 'GET',
+        },
+      },
+      {
+        http: {
+          path: 'users/friends',
+          method: 'GET',
+        },
+      },
+      {
+        http: {
+          path: 'users/is/this/a/long/path',
+          method: 'GET',
+        },
+      },
+    ];
+
+    return awsCompileApigEvents.compileResources().then(() => {
+      expect(awsCompileApigEvents.serverless.service.provider.compiledCloudFormationTemplate
+        .Resources.ApiGatewayResourceFoo).to.equal(undefined);
+      expect(awsCompileApigEvents.serverless.service.provider.compiledCloudFormationTemplate
+        .Resources.ApiGatewayResourceUsers).to.equal(undefined);
+      expect(awsCompileApigEvents.serverless.service.provider.compiledCloudFormationTemplate
+        .Resources.ApiGatewayResourceUsersFriends).to.equal(undefined);
+      expect(awsCompileApigEvents.serverless.service.provider.compiledCloudFormationTemplate
+        .Resources.ApiGatewayResourceUsersIsThis).to.equal(undefined);
+    });
+  });
+
+  it('should throw error if parent of existing resources is required', () => {
+    awsCompileApigEvents.serverless.service.provider.apiGateway = {
+      restApiId: '6fyzt1pfpk',
+      restApiRootResourceId: 'z5d4qh4oqi',
+      restApiResources: {
+        'users/friends': 'fcasdoojp1',
+      },
+    };
+
+    awsCompileApigEvents.validated.events = [
+      {
+        http: {
+          path: 'users',
+          method: 'GET',
+        },
+      },
+      {
+        http: {
+          path: 'users/friends/{friendId}',
+          method: 'GET',
+        },
+      },
+    ];
+
+    try {
+      return awsCompileApigEvents.compileResources()
```

Comment:
I might be wrong, but I think this could be written in a more straightforward way:

```js
expect(() => {
  return awsCompileApigEvents.compileResources();
}).to.throw(Error, 'Resource ID for path users is required');
```